### PR TITLE
feat(route_handler): better avoidance drivable areas extension in behavior path planning

### PR DESCRIPTION
**Pull request by zulfaqar-azmi-t4**
## Related Issue(required)

#234 

## Description(required)

This PR add adjacent lane check for left side and right side of input lanelet.
by implementing the check function, it is possible to check whether there exist any lane at the left side or right side of ego vehicle lane.

This PR is a port from .iv. It is divided into two parts, 

1. route_handler(no change to behavior)
2. behavior_path_planner (shift_length computation feature)

The feature implement adjacent lanelet search to prevent planner from generating avoidance path that extend towards non-lane areas like below

image

It also improve drivable area extension by appending lanelet that is nearby obstacle.

## Related PR(optional)
universe
#285 

this feature is tested in autoware.iv this scenario simulator
.iv
Azu : https://github.com/tier4/autoware.iv/pull/2383
Murooka : https://github.com/tier4/autoware.iv/pull/1839
https://github.com/tier4/AutonomousDrivingScenarios/pull/287

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [ ] Read [commit-guidelines][commit-guidelines]
- [ ] Assign PR to reviewer

If you are adding new package following items are required:

- [ ] Documentation with description of the package is available
- [ ] A sample launch file and parameter file are available if the package contains executable nodes

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[commit-guidelines]: https://www.conventionalcommits.org/en/v1.0.0/
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

**Comment by taikitanaka3** [comment]
@rej55 
can you take a look at this?

**Comment by taikitanaka3** [comment]
@zulfaqar-azmi-t4 
you need sign off commit to track committer

File: planning/route_handler/include/route_handler/route_handler.hpp [hunk 1]
```diff
@@ -163,8 +239,6 @@ class RouteHandler
   bool isBijectiveConnection(
     const lanelet::ConstLanelets & lanelet_section1,
     const lanelet::ConstLanelets & lanelet_section2) const;
-  bool getPreviousLaneletWithinRoute(
```
Comment (taikitanaka3) [inline planning/route_handler/include/route_handler/route_handler.hpp:166]: @zulfaqar-azmi-t4 
can you stay this line still to make less diff ?

**Comment by takayuki5168** [comment]
LGTM

File: planning/route_handler/include/route_handler/route_handler.hpp (hunk 1 above)
Comment (taikitanaka3) [reply planning/route_handler/include/route_handler/route_handler.hpp:166]: For the you moved to public is to use  this in for feature PR and it needs to be public. I resolve this issue.

**Review by taikitanaka3 — APPROVED** [review]
LGTM

**Review by rej55 — APPROVED** [review]
LGTM

**Pull request merged**